File: LeetCode/[M] - 49. Group Anagrams/solution.py
from typing import List


class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        group = {}

        for word in strs:
            wordS = "".join(sorted(list(word)))
            if wordS in group:
                group[wordS].append(word)
            else:
                group[wordS] = [word]

        return list(group.values())

File: LeetCode/[M] - 49. Group Anagrams/solution.go
package main

import "slices"

func groupAnagrams(strs []string) [][]string {
	m := make(map[string][]string)
	for _, word := range strs {
		bytes := []byte(word)
		slices.Sort(bytes)
		wordS := string(bytes)
		m[wordS] = append(m[wordS], word)
	}
	res := make([][]string, 0, len(m))
	for _, value := range m {
		res = append(res, value)
	}
	return res
}
